ardere/step_functions.py

```diff
@@ -368,15 +367,23 @@ def check_drained(self):
 
         """
         client = self.boto.client('ecs')
-        actives = len(
-            client.list_container_instances(
-                cluster=self.event["ecs_name"],
-                maxResults=1,
-                status="ACTIVE",
-            ).get('containerInstanceArns', []))
-        if actives:
+        actives = client.list_container_instances(
+            cluster=self.event["ecs_name"],
+            maxResults=1,
+            status="ACTIVE",
+        ).get('containerInstanceArns', [])
+        # filter out metric servers
+        if self.event["metrics_options"]["enabled"]:
```

Comment:
The shutdown plan already locates the metrics service, we should probably determine when we call it the first time and stick it in the event to avoid having to keep searching (and slowing execution time).